packages/samples/atomic-react/package.json

```diff
@@ -29,7 +29,7 @@
   "scripts": {
     "build": "tsc --noEmit && tsc --module commonjs --noEmit",
     "dev": "npm run build:assets && npm run dev:webpack",
-    "dev:webpack": "webpack serve",
+    "dev:webpack": "nx run atomic:build && webpack serve",
```

Comment:
Webpack would error if run without running build

Reply:
I recommend not to use the NX CLI to run parent tasks in scripts, since it doesn't add them to the dependency graph.

Instead, you should either:
* Modify AtomicReact's `project.json` to tell it that "dev" is dependent on building the parent packages. For example:
	```json
	{
	  "name": "atomic-react",
	  "targets": {
	    "dev": {
	      "dependsOn": ["^build"],
	      "inputs": [],
	      "outputs": [],
	      "executor": "nx:run-script",
	      "options": {
	        "script": "dev"
	      }
	    }
	  }
	}
	```
* Modify `nx.json` to require all `dev` tasks to build their parent task first. At the moment, I think this is better?

Reply:
Cool - good idea! But couple of things
* this would only work with `nx run` and not `npm run` (which is to be expected I guess, but still something that could be missed when looking at package.json scripts and running them)
* building all parent deps is cool - but in this case of atomic-react is builds beuno and headless too in addition to atomic which is what is needed. And there doesn't seem to be a cache / dependency change detection - the parent builds are run every time. Is this expected?

Reply:
it's expected. I did want to eventually add caching to Bueno and Headless but the story about Nx was starting to take too long so I left them as technical debt tickets.

Personally, I'd prefer if we choose one of those two extremes, no middle ground:
* Only build/debug AtomicReact, don't try to build Atomic.
* Build the whole stack (using Nx dependencies).